### DISCUSSION TITLE
Get current status of "jk-bms_charging" via MQTT ?

### DISCUSSION THREAD

**Discussion by umhh** [syssi/esphome-jk-bms#400] [Q&A]
Hi,
I use esphome without home assistant and found the proper mqtt-message in the discussion posts 
to turn on/off charging (publish ON/OFF to .../jk-bms_charging/command)

Is there a way to also get the current state of the charging or discharge switch via mqtt-message ?
While esphome sends out all kinds of other information on the BMS regularly via mqtt the current  status of the charging switch only seems to get sent on a status change.

**Comment by syssi** [comment]
There is a `/state` topic available per entity which holds the current state/measurement. Per default the messages are retained so you should find the last message on the broker even if the device is unavailable:

```
storage-sensors/sensor/storage-sensors_esphome_version/state 2023.12.5 Dec 26 2023, 17:23:12
storage-sensors/sensor/storage-sensors_uptime/state 1
storage-sensors/sensor/storage-sensors_temperature/state 9.4
storage-sensors/sensor/storage-sensors_pressure/state 955.8
storage-sensors/sensor/storage-sensors_humidity/state 70.4
storage-sensors/sensor/storage-sensors_illuminance/state 0.0
storage-sensors/sensor/storage-sensors_wifi_signal/state -36
storage-sensors/switch/storage-sensors_relay/state OFF
```

If an entity is writable the ESPHome node does listen on the dedicated `/command` topic for instructions. To track the state of a switch you should prefer the `/state` topic over the `/command` topic.

If you don't like retained messages you could force the sensor to be published periodically if you apply a heartbeat filter: 

```
filters:
  - heartbeat: 5s
```

Keep in mind this will lead to a lot of noise on your broker.
